packages/near-fast-auth-signer-e2e-tests/tests/devices.spec.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  KeyPair,
+} from '@near-js/crypto';
+import { expect, test } from '@playwright/test';
+
+import PageManager from '../pages/PageManager';
+import {
+  createAccount, deleteAccount, generateKeyPairs, initializeAdmin,
+  isServiceAccountAvailable
+} from '../utils/createAccount';
+import { getRandomEmailAndAccountId } from '../utils/email';
+import { setupPasskeysFunctions } from '../utils/passkeys';
+
+let testUserUid;
+
+test.beforeAll(async () => {
+  if (isServiceAccountAvailable()) {
+    initializeAdmin();
+  }
+});
+
+test('device page delete existing keys and continue sign in', async ({ page, baseURL }) => {
+  test.skip(!isServiceAccountAvailable(), 'Skipping test due to missing service account');
+
+  const pm = new PageManager(page);
+  test.setTimeout(120000);
+  const { email, accountId } = getRandomEmailAndAccountId();
+
+  await page.goto(baseURL);
+  const walletSelector = page.locator('#ws-loaded');
+  await walletSelector.waitFor();
+
+  const oidcKeyPair = KeyPair.fromRandom('ED25519');
+
+  // As of 14 May 2024, creating an account with 5 keypairs will be just enough to redirected to devices page
+  const keypairs = generateKeyPairs(5);
+  const {
+    createAccountResponse,
+    userUid
+  } = await createAccount({
+    email,
+    accountId,
+    oidcKeyPair,
+    keypairs,
+  });
+
+  // will be used to delete account
+  testUserUid = userUid;
+
+  expect(createAccountResponse.ok).toBe(true);
+
+  await setupPasskeysFunctions(page, 'page', {
+    isPassKeyAvailable:  true,
+    keyPairForCreation:  oidcKeyPair,
+    keyPairForRetrieval: oidcKeyPair,
+    shouldCleanStorage:  false
+  });
+
+  await pm.getLoginPage().signInWithEmail(email);
+  await page.waitForLoadState('domcontentloaded');
+
+  await pm.getAuthCallBackPage().handleEmail(email, [], {
+    isPassKeyAvailable:  true,
+    keyPairForCreation:  oidcKeyPair,
+    keyPairForRetrieval: oidcKeyPair,
+    shouldCleanStorage:  false
+  });
+
+  // Wait for page to render and execute async operations
+  await page.waitForTimeout(20000);
```

Comment:
I believe waiting for timeouts on tests are not a good practice. 

They expect a content to load on a certain time, and slow/fast connection/device will load the page on different times. 

It would be better to replace this with waiting for a network call to finish or content to be visible. 

Can we do that here?

Reply:
will replace it with recursive wait check so that it won't wait statically.

Reply:
Can we avoid using the recursive wait check and use native playwright code? 

I will try some options here and see if I can make it work :)